### PR TITLE
Added Google Style Python Docstrings to the module

### PR DESCRIPTION
**Pull request by iraquitan**
Added docstrings by following the [Google Style](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). For the docstrings I used the links provided by the README, but some descriptions can be improved.

### CODE REVIEW

File: pycnn.py [hunk 1]
```diff
@@ -41,16 +41,45 @@
...
+    """
 
     def f(self, x, t, Ib, Bu, tempA):
+        """Computes the derivative of x at t."""
         x = x.reshape((self.n, self.m))
         dx = -x + Ib + Bu + sig.convolve2d(self.cnn(x), tempA, 'same')
         return dx.reshape(self.m * self.n)
 
     def cnn(self, x):
+        """Approximate sigmoidal kernel function for the neural network."""
```
Comment (ankitaggarwal011) [inline pycnn.py]: This function is missing arguments description.
x is input to the piece-wise linear sigmoid function.
Please update the function description as "piece-wise linear sigmoid function".

File: pycnn.py [hunk 2]
```diff
@@ -41,16 +41,45 @@
...
+    Image Processing.
+
+
+    Attributes:
+        n (int): The number of rows from the image matrix.
+        m (int): The number of columns form the image matrix.
+    """
 
     def f(self, x, t, Ib, Bu, tempA):
+        """Computes the derivative of x at t."""
```
Comment (ankitaggarwal011) [inline pycnn.py]: This function is missing arguments description.
Ib is system bias.
Bu is convolution of control template with input.
tempA is feedback template.

File: pycnn.py [hunk 3]
```diff
@@ -41,16 +41,45 @@
...
+    original application of CNN processors was to perform real-time ultra-high
+    frame-rate (>10,000 frame/s) processing unachievable by digital processors.
+
+    This python library is the implementation of CNN for the application of
+    Image Processing.
+
+
+    Attributes:
+        n (int): The number of rows from the image matrix.
+        m (int): The number of columns form the image matrix.
```
Comment (ankitaggarwal011) [inline pycnn.py]: Additionally, n is the height and m is the width of the image.

**Review by ankitaggarwal011 — COMMENTED** [review]
Please look at the specific comments. Thanks.

File: pycnn.py [hunk 4]
```diff
@@ -67,6 +96,20 @@ def validate(self, input_location):
     # tempA: control template, tempB: feedback template
     def imageprocessing(self, inputlocation, outputlocation,
                         tempA, tempB, initialcondition, Ib, t):
+        """Process the image with the input arguments.
+
+        Args:
+            inputlocation (str): The string path for the input image.
+            outputlocation (str): The string path for the output processed
+                image.
+            tempA (:obj:`list` of :obj:`list`of :obj:`float`): The template
```
Comment (ankitaggarwal011) [inline pycnn.py]: Please refer tempA (template A) as the feedback template and tempB (template B) as control template everywhere.

The above comment " # tempA: control template, tempB: feedback template" is wrong, and I'll update it after merging this request. Thanks.

**Comment by ankitaggarwal011** [comment]
@iraquitan, I've added the init method to the object to initialize m and n variables. Would you please document it as well while making the other changes. Thanks a lot. Great work!

Please make sure your working on the updated version of the project.

**Comment by iraquitan** [comment]
Added the requested changes.

File: pycnn.py [hunk 5]
```diff
@@ -41,20 +41,62 @@
...
+            Bu: Convolution of control template with input.
+            tempA (:obj:`list` of :obj:`list`of :obj:`float`): Feedback
+                template.
+        """
         x = x.reshape((self.n, self.m))
         dx = -x + Ib + Bu + sig.convolve2d(self.cnn(x), tempA, 'same')
         return dx.reshape(self.m * self.n)
 
     def cnn(self, x):
+        """"Piece-wise linear sigmoid function.
```
Comment (ankitaggarwal011) [inline pycnn.py]: There is an extra quote here.

File: pycnn.py [hunk 6]
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
...
+
+        Args:
+            inputlocation (str): The string path for the input image.
+            outputlocation (str): The string path for the output processed
+                image.
+            tempA (:obj:`list` of :obj:`list`of :obj:`float`): Feedback
+                template.
+            tempB (:obj:`list` of :obj:`list`of :obj:`float`): Control
+                template.
+            initialcondition (float): The initial template state.
```
Comment (ankitaggarwal011) [inline pycnn.py]: initialcondition isn't related to template, you can simply refer it as "The initial state." Please change it at other places as well.

File: pycnn.py [hunk 7]
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
...
+            Ib (float): The template parameter z.
```
Comment (ankitaggarwal011) [inline pycnn.py]: Ib is system bias, the parameter z in a cellular neural network. Please change it at other places as well.

File: pycnn.py [hunk 8]
```diff
@@ -71,6 +113,20 @@ def validate(self, input_location):
...
+            t (numpy.ndarray): A numpy array with evenly spaced numbers.
```
Comment (ankitaggarwal011) [inline pycnn.py]: "A numpy array with evenly spaced numbers representing time points". Please change it at other places as well.

**Comment by ankitaggarwal011** [comment]
@iraquitan, you've done an excellent job. Kudos! There are only a few minor corrections and we're good to go. Thanks a lot for your contribution. I appreciate your help. :+1:

**Comment by iraquitan** [comment]
Added and fixed the requested changes.

**Comment by ankitaggarwal011** [comment]
Excellent work @iraquitan. Thank you for your contribution. Your help is much appreciated. :+1:

**Pull request merged**